plugins/synapticsmst/fu-plugin-synapticsmst.c

```diff
@@ -69,8 +69,11 @@ fu_plugin_synapticsmst_create_simple_guid (FuDevice *fu_device,
 					   const gchar *str)
 {
 	guint16 board_id = synapticsmst_device_get_board_id (device);
+	g_autofree gchar *guid = NULL;
 	g_autofree gchar *devid = g_strdup_printf ("MST-%s-%u", str, board_id);
 	fu_device_add_instance_id (fu_device, devid);
+	guid = fwupd_guid_hash_string (devid);
+	fu_device_add_guid (fu_device, guid);
```

Comment:
Na, `fu_device_convert_instance_ids()` should do this for us. It's not being called for some reason.

Reply:
* Plugins will only need to need to call this manually when adding child
     * devices, as fu_device_setup() automatically calls this after the
     * fu_device_probe() and fu_device_setup() virtual functions have been run.

Reply:
But there are not setup or probe vfuncs that will get called in synapticsmst at this time.

Reply:
Okay, my mistake, sorry. It would be a lot simpler (for me!) if `SynapticsMSTDevice` subclassed `FuDevice` like the other plugins. Could we do `fu_device_convert_instance_ids (fu_device)` instead here?

Reply:
I guess just manually calling `fu_device_setup()` before adding the device works too.  I just did that.

Reply:
I agree, subclassing would make it more like the rest but that's quite a bit of a plugin overhaul, so not any time soon..

Manually calling fu_device_setup() at that time does call fu_device_convert_instance_ids, so that should do the trick - I confirmed that works.